### PR TITLE
add GT_STRUCT_TUPLE

### PR DESCRIPTION
**Pull request by anstaf**

### CODE REVIEW

File: include/gridtools/common/tuple_util.hpp [hunk 1]
```diff
@@ -142,6 +149,60 @@
...
+    friend gt_##name##_tuple_getter tuple_getter(name);                                                               \
+    friend ::gridtools::meta::list<BOOST_PP_SEQ_ENUM(BOOST_PP_SEQ_TRANSFORM(GT_STRUCT_TUPLE_IMPL_TYPE_, _, members))> \
+        tuple_to_types(name);                                                                                         \
+    friend ::gridtools::meta::always<name> tuple_from_types(name)
+
+/*
+ * Struct adapter to tuple_like
+ *
+ * Usage:
+ * Declaring a struct like this make this struct tuple-like:
```
Comment (fthaler) [inline include/gridtools/common/tuple_util.hpp:181]:
```suggestion
 * Declaring a struct like this, makes it tuple-like:
```

File: include/gridtools/common/tuple_util.hpp [hunk 2]
```diff
@@ -142,6 +149,60 @@
...
+ * ```
+ * I.e. one can access the members both by name and by `get` accessor:
+ * ```
+ *    foo obj;
+ *    obj.a = 42;
+ *    assert(tuple_util::get<1>(obj) == 42);
+ * ```
+ * Also tuple algorithms works with `foo` as expected:
+ * ```
+ *    auto x = tuple_util::trasnform([](auto x) { return x * 2; }, foo{1, 2.5});
```
Comment (fthaler) [inline include/gridtools/common/tuple_util.hpp:199]:
```suggestion
 *    auto x = tuple_util::transform([](auto x) { return x * 2; }, foo{1, 2.5});
```

**Review by fthaler — APPROVED** [review]

**Comment by anstaf** [comment]
launch jenkins

**Comment by anstaf** [comment]
launch jenkins

**Comment by anstaf** [comment]
launch jenkins

**Comment by anstaf** [comment]
launch jenkins

**Pull request merged**